include/openrave/openravecontainer.h

```diff
@@ -0,0 +1,265 @@
+// -*- coding: utf-8 -*-
+// Copyright (C) 2024 OpenRAVE
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** \file openravecontainer.h
+    \brief Collection of high performance containers
+ */
+#ifndef OPENRAVE_CONTAINER_H
+#define OPENRAVE_CONTAINER_H
+
+#include <openrave/config.h>
+#include <openrave/openraveexception.h>
+
+#include <stdint.h>
+#include <vector>
+
+namespace OpenRAVE {
+
+namespace orcontainer {
+
+/// \brief Represents Key-value pair.
+template<typename DataType> struct NamedData
+{
+    NamedData(uint64_t nameId_, const DataType& data_) : nameId(nameId_), data(data_) {}
+
+    ///\brief invalidates this object
+    inline void Invalidate() { nameId = 0; }
+
+    ///\brief Checks validity of this object
+    ///\return whether valid
+    inline bool IsValid() const { return nameId > 0; }
+
+    ///\brief gets value
+    inline const DataType& GetValue() const { return data; }
+
+    ///\brief gets key
+    inline uint64_t GetKey() const { return nameId; }
+
+    ///\brief sets value for key
+    ///\param nameId_ key for the value
+    ///\param data_ value for the key
+    inline void Set(uint64_t nameId_, const DataType& data_)
+    {
+        nameId = nameId_;
+        data = data_;
+    }
+
+    uint64_t nameId = 0; ///< id for the semantic meaning of data. 0 means data is not valid.
+    DataType data; ///< user-defined data. If nameId is 0, invalid
+};
+
+/// \brief Associative data structure similar to map, optimized to avoid memory allocation. More suitable for relatively small number of elements.
+/// Similar to map, nameId(key) is unique. However, there can be multiple invalid nameIds.
+template<typename DataType> struct Map
+{
+    /// \brief gets number of valid entries
+    /// \return number of valid entries
+    inline size_t GetSize() const { return _numValidElements; }
+
+    /// \brief invalidates all entries while preserving memory
+    void Clear()
+    {
+        // Just invalidate each element, but keep memory allocated
+        for (NamedData<DataType>& customData : _vNamedDatas) {
+            customData.Invalidate();
```

Comment:
Inside the `Clear` function, you call `Invalidate`? Can you standardize on one name?